prow/cmd/deck/main.go

```diff
@@ -134,25 +133,56 @@ func main() {
 	}
 
 	// setup common handlers for local and deployed runs
-	mux.Handle("/", staticHandlerFromDir(o.staticFilesLocation))
+	mux.Handle("/static/", http.StripPrefix("/static", staticHandlerFromDir(o.staticFilesLocation)))
 	mux.Handle("/config", gziphandler.GzipHandler(handleConfig(configAgent)))
-	mux.Handle("/branding.js", gziphandler.GzipHandler(handleBranding(configAgent)))
 	mux.Handle("/favicon.ico", gziphandler.GzipHandler(handleFavicon(o.staticFilesLocation, configAgent)))
 	mux.Handle("/spyglass.js", gziphandler.GzipHandler(handleSpyglass(o.spyglass)))
 
-	if o.runLocal {
-		if o.spyglass {
-			initSpyglass(configAgent, o, mux, nil)
+	// Set up handlers for template pages.
+	mux.Handle("/pr", gziphandler.GzipHandler(handleSimpleTemplate(configAgent, "pr.html")))
+	mux.Handle("/command-help", gziphandler.GzipHandler(handleSimpleTemplate(configAgent, "command-help.html")))
+	mux.Handle("/plugin-help", http.RedirectHandler("/command-help", http.StatusMovedPermanently))
+	mux.Handle("/tide", gziphandler.GzipHandler(handleSimpleTemplate(configAgent, "tide.html")))
+	mux.Handle("/plugins", gziphandler.GzipHandler(handleSimpleTemplate(configAgent, "plugins.html")))
+
+	indexHandler := handleSimpleTemplate(configAgent, "index.html")
+	localDataHandler := staticHandlerFromDir("./localdata")
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		// This is necessary because a) it's impossible to set a handler for "/" that isn't a wildcard,
+		// and b) you cannot have multiple
+		if r.URL.Path != "/" {
+			if o.runLocal {
```

Comment:
nit: in go it would be fairly idiomatic to do this at init time when registering the handler, with a lambda we can return either a handler that does local or a handler that does non-local without the runtime dispatch